src/Hosting/Hosting/src/WebHostBuilderExtensions.cs

```diff
@@ -67,7 +71,7 @@ private static IWebHostBuilder Configure(this IWebHostBuilder hostBuilder, Actio
         /// <param name="hostBuilder">The <see cref="IWebHostBuilder"/> to configure.</param>
         /// <param name="startupFactory">A delegate that specifies a factory for the startup class.</param>
         /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
-        public static IWebHostBuilder UseStartup(this IWebHostBuilder hostBuilder, Func<WebHostBuilderContext, object> startupFactory)
+        public static IWebHostBuilder UseStartup<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]TStartup>(this IWebHostBuilder hostBuilder, Func<WebHostBuilderContext, TStartup> startupFactory)
```

Comment:
Why isn't this `[DynamicallyAccessedMembers(StartupAccessibility)]`?

Reply:
It could be but you have to pass a factory with an instance so constructors are handled in this case.

Reply:
Oh, because the `Func` creates the Startup object.